jaeger-core/src/main/java/com/uber/jaeger/samplers/ProbabilisticSampler.java

```diff
@@ -33,25 +33,38 @@
 public class ProbabilisticSampler implements Sampler {
   public static final String TYPE = "probabilistic";
 
-  private final long positiveSamplingBoundary;
-  private final long negativeSamplingBoundary;
-  private final double samplingRate;
-  private final Map<String, Object> tags;
+  private final double DELTA = 0.00001;
+  private long positiveSamplingBoundary;
+  private long negativeSamplingBoundary;
+  private double samplingRate;
+  private Map<String, Object> tags;
```

Comment:
Yeah, but it is wasteful to create and throw away up to 2000 probabilistic samplers every minute when we don't need to. That being said, I like immutability because it simplifies certain code. In this case, making all of the Samplers immutable will lead to easier to understand code.
Do you think we should do that at the expense of gc (which even at 2k per minute would only be noticed by very constrained applications)?